File: src/csharp/Intel/Iced/Intel/DecoderInternal/OpCodeHandlersTables_Legacy.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if DECODER
namespace Iced.Intel.DecoderInternal {
	static partial class OpCodeHandlersTables_Legacy {
		internal static readonly OpCodeHandler[] Handlers_MAP0;

		static OpCodeHandlersTables_Legacy() {
			var handlerReader = new LegacyOpCodeHandlerReader();
			var deserializer = new TableDeserializer(handlerReader, MaxIdNames, GetSerializedTables());
			deserializer.Deserialize();
			Handlers_MAP0 = deserializer.GetTable(Handlers_MAP0Index);
		}
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/dec/OpCodeHandlersTables_Legacy.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.dec;

import com.github.icedland.iced.x86.internal.ResourceReader;

final class OpCodeHandlersTables_Legacy {
	static final OpCodeHandler[] handlers_MAP0;

	static byte[] getSerializedTables() {
		return ResourceReader.readByteArray(OpCodeHandlersData_Legacy.class.getClassLoader(),
				"com/github/icedland/iced/x86/dec/OpCodeHandlersData_Legacy.bin");
	}

	static {
		LegacyOpCodeHandlerReader handlerReader = new LegacyOpCodeHandlerReader();
		TableDeserializer deserializer = new TableDeserializer(handlerReader, OpCodeHandlersData_Legacy.MAX_ID_NAMES, getSerializedTables());
		deserializer.deserialize();
		handlers_MAP0 = deserializer.getTable(OpCodeHandlersData_Legacy.HANDLERS_MAP0_INDEX);
	}
}
